repository/src/test/java/org/alfresco/repo/content/ContentServiceImplUnitTest.java

```diff
@@ -166,34 +166,46 @@ public void testRequestContentDirectUrl_StoreIsEnabledButNotImplemented()
     @Test
     public void testRequestContentDirectUrl_StoreRequestContentDirectUrlIsCalledWithFileNameOverride()
     {
+        DirectAccessUrl mockDirectAccessUrl = new DirectAccessUrl();
+        mockDirectAccessUrl.setAttachment(true);
+        mockDirectAccessUrl.setContentUrl(SOME_CONTENT_URL);
         final String fileNameOverride = "fileNameOverride.txt";
         setupSystemWideDirectAccessConfig(ENABLED);
         when(mockContentStore.isContentDirectUrlEnabled()).thenReturn(ENABLED);
+        when(mockContentStore.requestContentDirectUrl(anyString(), eq(true), eq(SOME_FILE_NAME), anyString(), anyLong())).thenReturn(mockDirectAccessUrl);
 
         DirectAccessUrl directAccessUrl = contentService.requestContentDirectUrl(NODE_REF, PROP_CONTENT_QNAME, true, 20L, fileNameOverride);
-        assertNull(directAccessUrl);
+        assertEquals("fileName was not set properly on the DirectAccessUrl response", fileNameOverride, directAccessUrl.getFileName());
```

Comment:
"fileName was not set properly on the DirectAccessUrl response" could be a private constant in this test class as this particular string is repeated 3 times.